### PR TITLE
Undeprecate typed yarpcerror creation and validation functions

### PR DESCRIPTION
**Pull request by willhug**
Summary: #1303 ended up adding new yarpc error apis and deprecating
these convenience functions for creating errors.  I think the overall
direction of #1303 is better in general but I think the convenience of
these functions should not be understated.  Instead of writing

  yarpcerrors.Newf(yarpcerrors.CodeInternal, "test")

I want to write:

  yarpcerrors.InternalErrorf("test")

Which <in my opinion> is a much cleaner api.

Full disclosure, I have a lot of internal packages that are using this
API in tests/etc.  And internally, using a deprecated API will cause
your lint to fail... so there's that.

In terms of what we encourage customers to use, I'm pretty flexible, but
I think the api has a lot of value, and it's frustrating to be
discouraged from using it.

### CODE REVIEW

File: yarpcerrors/errors.go [hunk 1]
```diff
@@ -159,227 +159,163 @@ func ErrorMessage(err error) string {
...
+	return Status(err).Code() == CodeDataLoss
 }
 
-// IsUnauthenticated returns true if ErrorCode(err) == CodeUnauthenticated.
-//
-// Deprecated: Use FromError and Code instead.
+// IsUnauthenticated returns true if Status(err).Code() == CodeUnauthenticated.
 func IsUnauthenticated(err error) bool {
-	return ErrorCode(err) == CodeUnauthenticated
+	return Status(err).Code() == CodeUnauthenticated
```
Comment (peter-edge) [inline yarpcerrors/errors.go]: `Status` is the struct type, this is a cast, I think what you meant was `FromError(err).Code()`

**Comment by peter-edge** [comment]
Oh obviously get the tests passing ie `s/Status(err)/FromError(err)/g` on this diff

**Comment by peter-edge** [comment]
@willhug I fixed the `Status/FromError` thing, I also moved the deprecated functions to the bottom and updated the docs for the `*Errorf` functions to indicate what they are doing. There was some messup with #1313 that the changes there got pulled in so I manually reverted them, my bad.

tl;dr this should be ready to go.
